src/predictions/rudderstack_predictions/py_native/llm.py

```diff
@@ -23,11 +24,35 @@ class LLMModel(BaseModelType):
 
     def __init__(self, build_spec: dict, schema_version: int, pb_version: str) -> None:
         super().__init__(build_spec, schema_version, pb_version)
+        if build_spec["llm_model_name"] == "":
```

Comment:
You are making the assumption if no value is provided it would be an empty string, which is not correct.

```suggestion
        if build_spec.get("llm_model_name", None) is None:
```

Reply:
Umm then if the user does not put anything in the llm_model_name then it will itself through an error.

Reply:
But I tried it and it shows me this error.
2024-05-06T10:25:21.515+0530    ERROR   compile/compile.go:55   creating project api: loading project: could not create project: failed to read project yaml 
Error: validating project sample_attribution: getting models for folder: models: error listing models: error parsing model build spec for sf_cortex_test: build spec is invalid: [llm_model_name: Invalid type. Expected: string, given: null]

Reply:
ok then I will remove the empty string check